crates/re_viewport/src/space_view_entity_picker.rs

```diff
@@ -168,25 +162,21 @@ fn add_entities_line_ui(
     entity_tree: &EntityTree,
     space_view: &SpaceViewBlueprint,
     query_result: &DataQueryResult,
-    inclusions: &HashSet<EntityPathExpr>,
-    exclusions: &HashSet<EntityPathExpr>,
+    entity_path_filter: &EntityPathFilter,
     entities_add_info: &IntMap<EntityPath, EntityAddInfo>,
 ) {
     ui.horizontal(|ui| {
         let entity_path = &entity_tree.path;
 
         let add_info = entities_add_info.get(entity_path).unwrap();
 
-        let included = query_result.contains_any(entity_path);
-
-        // TODO(jleibs): Speed this up
-        let excluded = exclusions.iter().any(|expr| match expr {
-            EntityPathExpr::Exact(expr) => expr == entity_path,
-            EntityPathExpr::Recursive(expr) => entity_path.starts_with(expr),
-        });
+        let is_explicitly_excluded = entity_path_filter.is_explicitly_excluded(entity_path);
+        let is_explicitly_included = entity_path_filter.is_explicitly_included(entity_path);
+        let is_included = query_result.contains_any(entity_path);
+        let is_included = entity_path_filter.is_included(entity_path);
```

Comment:
Not sure which one to use here. I guess they should be equivalent, barring bugs?

Reply:
I'm pretty sure `entity_path_filter` is the right one to add here because it's the entity-filter-editor. It's possible we end up using entity_path_filters in a future context where we don't necessarily have a query.